src/syft/lib/python/slice.py

```diff
@@ -0,0 +1,204 @@
+# stdlib
+from typing import Any
+from typing import Iterable
+from typing import Optional
+from typing import Union
+
+# third party
+from google.protobuf.reflection import GeneratedProtocolMessageType
+
+# syft relative
+from ... import deserialize
+from ... import serialize
+from ...core.common import UID
+from ...core.common.serde.serializable import bind_protobuf
+from ...logger import traceback_and_raise
+from ...logger import warning
+from ...proto.lib.python.slice_pb2 import Slice as Slice_PB
+from .iterator import Iterator
+from .primitive_factory import PrimitiveFactory
+from .primitive_factory import isprimitive
+from .primitive_interface import PyPrimitive
+from .types import SyPrimitiveRet
+from .util import downcast
+from .util import upcast
+
+
+@bind_protobuf
+class Slice(int, PyPrimitive):
+
+    def __init__(
+        self,
+        start: Any = None,
+        stop: Any = None,
+        step: Any = None,
+        id: Optional[UID] = None
+    ):
+        if not start and not stop and not step:
+            start = NULL
+            stop = NULL
+            step = NULL
```

Comment:
I am not sure why you have used `NULL` instead of `None`. Any specific reason? If not replace it with `None` and that should solve following logs 
```
src/syft/lib/python/slice.py:38:21: F821 undefined name 'NULL'
            start = NULL
                    ^
src/syft/lib/python/slice.py:39:20: F821 undefined name 'NULL'
            stop = NULL
                   ^
src/syft/lib/python/slice.py:40:20: F821 undefined name 'NULL'
            step = NULL
```

and some more similar errors, have a look